JSpecify: support a generic method call with explicit type arguments

According to the [JSpecify Spec](https://jspecify.dev/docs/user-guide/#generics), it is possible to add `extends @Nullable Object` to type parameters to siginify that `T` can be a nullable type.

Consider the following sample:

```java
import org.jspecify.annotations.*;

@NullMarked
public class Todo {
    public static <T extends @Nullable Object> T foo(NullableSupplier<T> code) {
        return code.get();
    }

    public static void main(String[] args) {
        Todo.<@Nullable Object> foo(() -> null);
    }

    @NullMarked
    @FunctionalInterface
    public interface NullableSupplier<T extends @Nullable Object> {
        T get();
    }
}
```

According to JSpecify Spec, this is valid since the call to `foo` in `main` explicitly specifies the type parameter `T` as **nullable** Object. The [JSpecify Reference Checker](https://github.com/jspecify/jspecify-reference-checker) also accepts this. 
NullAway however falsly gives us the following error without JSpecifyMode:

```
Todo.java:13: error: [NullAway] returning @Nullable expression from method with @NonNull return type
        Todo.<@Nullable Object> foo(() -> null);
                                    ^
    (see http://t.uber.com/nullaway )
1 error
```

And this error with JSpecifyMode:

```
Todo.java:13: error: [NullAway] Cannot pass parameter of type NullableSupplier<@Nullable Object>, as formal parameter has type NullableSupplier<T>, which has mismatched type parameter nullability
        Todo.<@Nullable Object> foo(() -> null);
                                    ^
    (see http://t.uber.com/nullaway )
1 error
```



This issue seems similar to #660 and others, but I feel like my example gives a proper way to verify that this is a Bug and that the Bug would be resolved if my sample compiles.

--- a/nullaway/src/test/java/com/uber/nullaway/jspecify/GenericMethodTests.java
+++ b/nullaway/src/test/java/com/uber/nullaway/jspecify/GenericMethodTests.java
@@ -161,6 +161,35 @@ public class GenericMethodTests extends NullAwayTestsBase {
         .doTest();
   }
 
+  @Test
+  public void issue1035() {
+    makeHelper()
+        .addSourceLines(
+            "Todo.java",
+            "import org.jspecify.annotations.*;",
+            "@NullMarked",
+            "public class Todo {",
+            "    public static <T extends @Nullable Object> T foo(NullableSupplier<T> code) {",
+            "        return code.get();",
+            "    }",
+            "    public static void main(String[] args) {",
+            "        // BUG: Diagnostic contains: returning @Nullable expression from method with @NonNull return type",
+            "        Todo.<Object>foo(() -> null);",
+            "        Todo.<@Nullable Object>foo(() -> null);",
+            "    }",
+            "    // this method should have no errors once we support inference for generic methods",
+            "    public static void requiresInferenceSupport() {",
+            "        // BUG: Diagnostic contains: returning @Nullable expression from method with @NonNull return type",
+            "        Todo.foo(() -> null);",
+            "    }",
+            "    @FunctionalInterface",
+            "    public interface NullableSupplier<T extends @Nullable Object> {",
+            "        T get();",
+            "    }",
+            "}")
+        .doTest();
+  }
+
   private CompilationTestHelper makeHelper() {
     return makeTestHelperWithArgs(
         Arrays.asList(

